src/clients/java/src/main/java/com/tigerbeetle/NativeClient.java

```diff
@@ -15,61 +14,28 @@ final class NativeClient implements AutoCloseable {
      * https://docs.oracle.com/javase%2F9%2Fdocs%2Fapi%2F%2F/java/lang/ref/Cleaner.html
      */
     private static final class NativeHandle implements Runnable {
-        // Keeping the contextHandle and a reference counter guarded by
-        // atomics in order to prevent the client from using a disposed
-        // context during `close()`.
         private long handle;
-        private final AtomicLong atomicHandleReferences;
-        private final AtomicLong atomicHandleClosePending;
-
-        private static final long REF_CLOSED = 1 << 0;
-        private static final long REF_ACCESS = 1 << 1;
 
         public NativeHandle(long handle) {
+            assert handle != 0;
             this.handle = handle;
-            this.atomicHandleReferences = new AtomicLong(0);
-            this.atomicHandleClosePending = new AtomicLong(0);
         }
 
-        // Silence PMD hint on `if (x) { if (y) {} }`. It is cleaner than `if (x && y)` here.
-        @SuppressWarnings("PMD.CollapsibleIfStatements")
-        public void submit(final Request<?> request) throws Exception {
-            // Bump refs to access. Bail if REF_CLOSED bit was set.
-            // After observing REF_CLOSED, any other modifications which dont unset it don't matter.
-            if ((atomicHandleReferences.addAndGet(REF_ACCESS) & REF_CLOSED) != 0) {
+        public synchronized void submit(final Request<?> request) throws Exception {
```

Comment:
Some considerations here:

1. I think that [double-checked locking](https://en.wikipedia.org/wiki/Double-checked_locking) will not save us much in this case. So, the C# implementation can be changed to match the other clients.

2. As library authors, we should not use `synchronized` methods in Java. A misused/malicious `lock` on the client instance could interfere with the intended behavior.

Instead use a private instance as a lock:

```java
private final Object lock = new Object();

public void submit() {
    synchronized(lock) {
        ...
    }
}
```

The C# implementation can use the same technique, but in C#, this pattern is called `syncRoot`.

```c#
private readonly object syncRoot = new object();

public void Submit()
{
    lock(syncRoot )
    {
         ...
    }
}
```